pkg/backup/volume_snapshot_action.go

```diff
@@ -38,11 +39,15 @@ type volumeSnapshotAction struct {
 
 var _ Action = &volumeSnapshotAction{}
 
-func NewVolumeSnapshotAction(snapshotService cloudprovider.SnapshotService) Action {
+func NewVolumeSnapshotAction(snapshotService cloudprovider.SnapshotService) (Action, error) {
```

Comment:
the intent was to embed the nil-checks into the library so if someone uses it outside the ark server they still get argument validation. what would you do differently? leave as it was?